src/python_testing/TC_SC_4_3.py

```diff
@@ -0,0 +1,407 @@
+        service_name = f'_I{compressed_fabric_id:016X}._sub.{MdnsServiceType.OPERATIONAL.value}'
+        return service_name
+
+    @staticmethod
+    def verify_decimal_value(input_value, max_value: int):
+        try:
+            input_float = float(input_value)
+            input_int = int(input_float)
+
+            if str(input_value).startswith("0") and input_int != 0:
+                return (False, f"Input ({input_value}) has leading zeros.")
+
+            if input_float != input_int:
+                return (False, f"Input ({input_value}) is not an integer.")
+
+            if input_int <= max_value:
+                return (True, f"Input ({input_value}) is valid.")
+            else:
+                return (False, f"Input ({input_value}) exceeds the allowed value {max_value}.")
+        except ValueError:
+            return (False, f"Input ({input_value}) is not a valid decimal number.")
+
+    def verify_t_value(self, t_value):
+        # Verify t_value is a decimal number without leading zeros and less than or equal to 6
+        try:
+            T_int = int(t_value)
+            if T_int < 0 or T_int > self.MAX_T_VALUE:
+                return False, f"T value ({t_value}) is not in the range 0 to 6. ({t_value})"
+            if str(t_value).startswith("0") and T_int != 0:
+                return False, f"T value ({t_value}) has leading zeros."
+            if T_int != float(t_value):
+                return False, f"T value ({t_value}) is not an integer."
+
+            # Convert to bitmap and verify bit 0 is clear
+            if T_int & 1 == 0:
+                return True, f"T value ({t_value}) is valid and bit 0 is clear."
+            else:
+                return False, f"Bit 0 is not clear. T value ({t_value})"
+        except ValueError:
+            return False, f"T value ({t_value}) is not a valid integer"
+
+    @staticmethod
+    def is_ipv6_address(addresses):
+        if isinstance(addresses, str):
+            addresses = [addresses]
+
+        for address in addresses:
+            try:
+                # Attempt to create an IPv6 address object. If successful, this is an IPv6 address.
+                ipaddress.IPv6Address(address)
+                return True, "At least one IPv6 address is present."
+            except ipaddress.AddressValueError:
+                # If an AddressValueError is raised, the current address is not a valid IPv6 address.
+                return False, f"Invalid IPv6 address encountered: {address}, provided addresses: {addresses}"
+        return False, "No IPv6 addresses found."
+
+    @staticmethod
+    def extract_ipv6_address(text):
+        items = text.split(',')
+        return items[-1]
+
+    @staticmethod
+    def verify_hostname(hostname: str) -> bool:
+        # Remove '.local' or '.local.' suffix if present
+        if hostname.endswith('.local'):
+            hostname = hostname[:-6]
+        elif hostname.endswith('.local.'):
```

Comment:
Does this imply that you might also get one that end in a dot without local?
ie, could you get _matter._tcp. ? If so, it might be easier to just remove the last dot, then remove .local

Reply:
Only local string, updated method with suggestion.